src/test/java/JsonUtilityTest.java

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.gson.Gson;
+import com.google.sps.utility.JsonUtility;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Test Json Utility methods */
+@RunWith(JUnit4.class)
+public class JsonUtilityTest {
```

Comment:
Are there any failure cases you need to test? What if you pass in a `null` value? What if the value is not a supported type, like some custom Java class?

Reply:
Yes, I should absolutely be testing failure and edge cases. Thanks for pointing this out!

Reply:
I have added two edge cases: `null` and my own custom Java class but I am having trouble finding an unsupported type which causes `sendJson` to fail. Let me know if you have any ideas.

Reply:
Thanks for the new cases. Don't worry about finding an error case - it's really something that Gson should take care of.

Reply:
Awesome, thanks!